src/libraries/System.Net.Security/src/System/Net/Security/SslStreamCertificateContext.Linux.cs

```diff
@@ -211,6 +211,20 @@ private SslStreamCertificateContext(X509Certificate2 target, ReadOnlyCollection<
                 return null;
             }
 
+            // see logic in AddRootCertificate for why this is necessary
+            IntPtr[] issuerHandles = ArrayPool<IntPtr>.Shared.Rent(IntermediateCertificates.Count == 0 ? 1 : IntermediateCertificates.Count);
+            if (IntermediateCertificates.Count == 0)
+            {
+                issuerHandles[0] = issuer;
+            }
+            else
+            {
+                for (int i = 0; i < IntermediateCertificates.Count; i++)
+                {
+                    issuerHandles[i] = IntermediateCertificates[i].Handle;
```

Comment:
`IntermediateCertificates` is a public property. Are there concerns around liveliness of these handles, or making sure we don't pass a null handle down to native code if another thread disposed of these certificates?

We have special handling for `caCert` above but not for these.

Reply:
Wouldn't there be a potential race between us reading Handle and potential .Dispose() call? Before reading your comment I added private `X509Certificate[]` to make sure the certs are kept alive

```cs
            _privateIntermediateCertificates = new X509Certificate2[intermediates.Count];
            for (int i = 0; i < intermediates.Count; i++)
            {
                _privateIntermediateCertificates[i] = new X509Certificate2(intermediates[i]);
            }
```

Or do we not care about this sort of problems?

Reply:
> Wouldn't there be a race

Yeah, and I was thinking about that when I wrote the comment and was trying to think of how it would really manifest (other than finalization from a GC hole).  So where I initially thought your redundant instantiations of the cert were excessive, I changed my mind to decide "hey, and it solves the potential ToC/ToU race with a manual call to Dispose".

Though it occurs to me now that you've created new disposable objects owned by this object, so now you really should be disposing them in this object's Dispose.  Except it doesn't have one, and you'd have to introduce thread safety again... so, nevermind.  Finalization enqueuing it is!